pkg/gardenlet/controller/backupbucket/backup_bucket_actuator.go

```diff
@@ -95,18 +100,35 @@ func (a *actuator) Reconcile(ctx context.Context) error {
 
 func (a *actuator) Delete(ctx context.Context) error {
 	var (
-		g                  = flow.NewGraph("Backup bucket deletion")
+		g = flow.NewGraph("Backup bucket deletion")
+
+		_ = g.Add(flow.Task{
+			Name: "Destroying generated backup bucket secret in garden cluster",
+			Fn:   a.deleteGeneratedBackupBucketSecretInGarden,
+		})
+		deployBackupBucketSecret = g.Add(flow.Task{
+			Name: "Deploying backup bucket secret to seed",
+			Fn:   flow.TaskFn(a.deployBackupBucketExtensionSecret).RetryUntilTimeout(defaultInterval, defaultTimeout),
+		})
+		deleteBackupBucketExtensionSecret = g.Add(flow.Task{
+			Name:         "Deleting backup bucket secret in seed",
+			Fn:           flow.TaskFn(a.deleteBackupBucketExtensionSecret).RetryUntilTimeout(defaultInterval, defaultTimeout),
```

Comment:
Hm, in this step don't we delete the just redeployed secret? If yes, I am wondering how the redeploying the secret helps as we delete it right away.

Reply:
The redeployment updates the credentials in case they were changed. This step here sets the deletion timestamp (finalizer protects it from being removed). I didn't change the order, but I agree that it's confusing. I will move it down after the deletion of the bucket resource to make it more clear.